spyplane/listeners.py

```diff
@@ -58,13 +58,18 @@ async def on_raw_reaction_add(payload: RawReactionActionEvent):
         return
     message = await bot.channel.fetch_message(payload.message_id)
     message.delete()
-    asyncio.create_task(record.record_reaction(message, payload.member.name, payload.member.id)) # Another option is to try a Queue
+    asyncio.create_task(record.record_reaction(message, payload.member.name, payload.member.id))  # Another option is to try a Queue
+
 
 @bot.event
 async def on_message(msg):
-    if msg.author.id==bot.user.id:
-        # print("Message from bot to itself")
-        pass
-    elif msg.author.id==BGS_BOT_USER_ID and msg.channel.id==TICK_CHANNEL:
-        print("Message from BGS Bot in tick channel!")
-        await PostAfterTickService().validate_message(msg)
+    try:
+        if msg.author.id==bot.user.id:
+            # print("Message from bot to itself")
+            pass
+        elif msg.author.id==BGS_BOT_USER_ID and msg.channel.id==TICK_CHANNEL:
+            log("Message from BGS Bot in tick channel!")
+            await PostAfterTickService().validate_message(msg)
+    except Exception as e:
```

Comment:
Normally this is true. 
However if you do not catch ALL exceptions here, then discord will call the `onerror` event handler, which does not provide any debug information on what the actual error was. 

Try removing the catch all and see the onerror handler being called - the original cause is lost - so we will be left unable to debug.

Reply:
Sounds like we should set up an on_error handler to catch/deal with those cases. Catching everything is generally a bad idea and will cause you potential other issues if you d follow that pattern.

Reply:
As mentioned in the previous comment, `on_error` is [already setup ](https://github.com/PilotsTradeNetwork/SpyPlane/blob/2b7c91cc2559409eaa68f566a5dd465d7536d0bf/spyplane/listeners.py#L38) and it's payload does not provide the cause, when an exception happens.

The only way to identify the cause of any error that happens `on_message` is to catch and log it. Otherwise the framework that calls the `on_message` will receive the exception, and swallow it without any message. 

Try adding this on your local to see what I mean.
```
import json
json.dumps(msg)
```
This will cause an exception `Message cannot be serialized to json`. But this will not show up anywhere, unless a catch all is added by us.

Reply:
Can you not forcefully log the exception during on_error to avoid the scenario of not knowing what went wrong then?

I still believe that you should catch the exceptions you expect here, rather than catching everything. Catching everything means you are not aware of what the underlying workflow is, and you could end up catching things that you really should not be catching, causing unintended scenarios.

Reply:
A generic catch all, as you say, can catch unexpected exceptions. Here the need **is to catch all unexpected exceptions**, and log them. Logging is crucial to investigation and debugging when something goes wrong.

Here is a short video: https://drive.google.com/file/d/1zfSOB57x-CAazGQEyedaaIq2N58qqMwS/view?usp=sharing

If you are able to cause an error in on_message, and log the real cause in on_error, I'm happy to admit I'm wrong, until then what you are saying shows that you did not understand the context nor read the past messages with attention